lint/private/lint_aspect.bzl

```diff
@@ -13,9 +13,21 @@ fail_on_violation_flag = rule(
     build_setting = config.bool(flag = True),
 )
 
-def report_file(mnemonic, target, ctx):
-    report = ctx.actions.declare_file("{}.{}.aspect_rules_lint.report".format(mnemonic, target.label.name))
-    return report, OutputGroupInfo(rules_lint_report = depset([report]))
+# buildifier: disable=function-docstring
+def report_files(mnemonic, target, ctx):
+    outfile = "{}.{}.aspect_rules_lint.{}"
+    report = ctx.actions.declare_file(outfile.format(mnemonic, target.label.name, "report"))
+    outs = [report]
+    if ctx.attr._options[LintOptionsInfo].fail_on_violation:
+        # Fail on violation means the exit code is reported to Bazel as the action result
+        exit_code = None
```

Comment:
would it be useful in the future to also capture the exit code with `fail_on_violation` is true so it can be reported as "Linter X failed with exit code Y"?

Reply:
there's nothing to capture in that case because it's either zero, or the action failed to run so it produces no outputs

Reply:
right. makes sense